src/blocks/net.rs

```diff
@@ -991,13 +991,18 @@ fn get_iw_ssid(dev: &NetworkDevice) -> Result<Option<String>> {
     }
 
     let raw = raw.unwrap();
-    let result = raw
-        .stdout
-        .split(|c| *c == b'\n')
-        .filter_map(|x| IW_SSID_REGEX.find(x))
-        .next();
+    let bytes = raw.stdout.split(|c| *c == b'\n').next().unwrap();
+    let cap = IW_SSID_REGEX.captures(bytes);
+    if cap.is_none() {
+        return Ok(None);
+    }
+    let match_obj = cap.unwrap().get(1);
+    if match_obj.is_none() {
```

Comment:
Also here.

Reply:
i couldn't find a way to do it more elegantly.
i needed to convert the match object to to a `&[u8]`, so the best way i saw was to retrieve it out of the Option, and call as_bytes() on it. And to be able to do that, i similarly had to pull out the captures out of the option.

I did verify though, that just calling the same processing chain, using unwrap() and removing the is_none() checks, and passing the result to maybe_ssid_convert, leads to a panic in case the output from iw does not match the expected format

Reply:
that results in an error on the line below:
```
error[E0599]: no method named `as_bytes` found for struct `regex::re_bytes::Captures<'_>` in the current scope
   --> src/blocks/net.rs:999:41
    |
999 |     maybe_ssid_convert(result.map(|x| x.as_bytes()))
    |                                         ^^^^^^^^ method not found in `regex::re_bytes::Captures<'_>`
```

Reply:
Ah yeah, forgot about that.
```
let result = raw
        .stdout
        .split(|c| *c == b'\n')
        .filter_map(|x| IWCTL_SSID_REGEX.captures_iter(x).next())
        .filter_map(|x| x.get(1))
        .next();
```

Reply:
@themadprofessor very interesting. thanks for sharing. 
so, basically use filter_map as a way to "absorb" intermediate None values and turn them into a "final" None value through next().

Interestingly, in the go world we tend to favor simple code even when it's a bit more verbose, 
but if the prevalent style in rust is to favor a bit more magic to create more concise code, that's fine with me too.
I'm happy to add a commit that changes it to your style. is that what @ammgws would prefer?

Reply:
I'm fine with it

Reply:
you're fine with the current code in this PR or with @themadprofessor 's version?
aside from "fine", is there a concensus which is "better"?
Also @ammgws seems you're a de facto maintainer, perhaps you should be added to https://github.com/greshake/i3status-rust/blob/master/CONTRIBUTING.md#maintainership ?

Reply:
I might update that sometime but I haven't had much time for this repo lately

Reply:
@Dieterbe Rust has a [WIP style guide](https://doc.rust-lang.org/1.0.0/style/), but generally if [Clippy](https://github.com/rust-lang/rust-clippy) is happy, its good code. Personally, I try to offload as much to the stdlib and the compiler since they normally know best.

Reply:
couldn't find much in the style guide about this specific tradeoff of using map_filter + next for processing Option's;
but since @themadprofessor 's suggestion definitely seems more in line with the rest of the code in general, i've applied the tweak in a new commit.
does this look good to merge now?